src/astro/sql/operators/load_file.py

```diff
@@ -0,0 +1,112 @@
+from typing import Any, Dict, Optional
+
+from airflow.hooks.base import BaseHook
+from airflow.models import BaseOperator
+from airflow.models.xcom_arg import XComArg
+
+from astro.constants import DEFAULT_CHUNK_SIZE, LoadExistStrategy
+from astro.databases import BaseDatabase, create_database
+from astro.files import File, get_files
+from astro.sql.table import Table
+from astro.utils.load import populate_normalize_config
+from astro.utils.task_id_helper import get_task_id
+
+
+class LoadFile(BaseOperator):
+    """Load S3/local table to postgres/snowflake database
+
+    :param input_file: File path and conn_id for object stores
+    :param output_table: Table to create
+    :param ndjson_normalize_sep: separator used to normalize nested ndjson.
+    :param chunk_size: Specify the number of records in each batch to be written at a time.
+    :param if_exists: Overwrite file if exists. Default False.
+    """
+
+    template_fields = ("output_table", "input_file")
+
+    def __init__(
+        self,
+        input_file: File,
+        output_table: Table,
+        chunk_size: int = DEFAULT_CHUNK_SIZE,
+        if_exists: LoadExistStrategy = "replace",
+        ndjson_normalize_sep: str = "_",
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.output_table: Table = output_table
+        self.input_file = input_file
+        self.chunk_size = chunk_size
+        self.kwargs = kwargs
+        self.if_exists = if_exists
+        self.ndjson_normalize_sep = ndjson_normalize_sep
+        self.normalize_config: Dict[str, str] = {}
+
+    def execute(self, context: Any) -> Table:
+        """
+        Load an existing dataset from a supported file into a SQL table.
+        """
+        if self.input_file.conn_id:
+            BaseHook.get_connection(self.input_file.conn_id)
```

Comment:
Why do we have these lines?

Reply:
to test if the conn_id exists or not.

Reply:
We should add this as a comment, also, it feels doesn't feel like a good way to check! We probably should add a method in OSS Airflow too to check if a Connection exists or not.

Until then though, we should do the following:

```suggestion
        if self.input_file.file_conn_id:
            try:
                BaseHook.get_connection(self.file_conn_id)
            except AirflowNotFoundException:
                # TODO: We should use our custom exception here
                raise AirflowNotFoundException(
                    f"Connection ('{self.input_file.conn_id}') does not exist in Airflow. "
                    "Please pass a valid connection ID to File()"
                )
```

We can even have a separate method/function:

```python
        def connection_exists(conn_id: str) -> bool
            try:
                BaseHook.get_connection(self.file_conn_id)
                return True
            except AirflowNotFoundException:
                return False
```